components/parser/src/main/kotlin/io/github/composegears/valkyrie/parser/PackageExtractor.kt

```diff
@@ -0,0 +1,39 @@
+package io.github.composegears.valkyrie.parser
+
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.invariantSeparatorsPathString
+
+object PackageExtractor {
+
+    private val knownSourceSets = listOf(
+        "src/main/kotlin",
+        "src/main/java",
+        "src/androidMain/kotlin",
+        "src/commonMain/kotlin",
+        "src/desktopMain/kotlin",
+        "src/jvmMain/kotlin",
+        "src/jsMain/kotlin",
+        "src/iosMain/kotlin",
+        "src/wasmJsMain/kotlin",
+        "src/linuxMain/kotlin",
+        "src/macosMain/kotlin",
+        "src/nativeMain/kotlin",
+    )
+
+    fun getFrom(path: String): String? {
+        val invariantPath = path.toPath().invariantSeparatorsPathString
```

Comment:
```suggestion
        val invariantPath = Path(path).invariantSeparatorsPathString
```

Reply:
But I would still suggest processing this step in the outsides.

Reply:
Could you please take a look?
https://github.com/ComposeGears/Valkyrie/compare/2caa814b2d8a5201411d5d6643aa09e881393b2e..ef5f451e30aee9158a01fe9a1d109a1a5882fc5e